kubernetes-contexts.el

```diff
@@ -78,6 +78,63 @@
    (display-buffer
     (kubernetes-yaml-make-buffer kubernetes-display-config-buffer-name config))))
 
+(defun kubernetes-contexts--context-names (state)
+  "Get a list of all available kubectl contexts from STATE."
+  (-let* ((config (or (kubernetes-state--get state 'config) (kubernetes-kubectl-await-on-async kubernetes-props state #'kubernetes-kubectl-config-view)))
+          ((&alist 'contexts contexts) config))
+    (--map (alist-get 'name it) contexts)))
+
+(defun kubernetes-contexts-use-context (context)
+  "Switch Kubernetes context refresh the pods buffer.
+
+CONTEXT is the name of a context as a string."
+  (interactive (list (completing-read "Context: " (kubernetes-contexts--context-names (kubernetes-state)) nil t)))
+  (kubernetes-process-kill-polling-processes)
+
+  (let ((state (kubernetes-state)))
+    (kubernetes-state-clear)
+    (kubernetes-state-update-overview-sections (kubernetes-state-overview-sections state)))
+
+  (kubernetes-state-trigger-redraw)
+
+  (when-let (buf (get-buffer kubernetes-overview-buffer-name))
+    (with-current-buffer buf
+      (goto-char (point-min))))
+
+  (let ((state (kubernetes-state)))
+    (kubernetes-kubectl-config-use-context
+     kubernetes-props
+     state
+     context
+     (lambda (_)
+       (when kubernetes-default-overview-namespace
+         (kubernetes-set-namespace kubernetes-default-overview-namespace
+                                   state))
+       (kubernetes-state-trigger-redraw)))))
+
+(defun kubernetes-contexts-rename (context new-name)
+  "Renames CONTEXT to NEW-NAME."
+  (interactive
+   (let* ((contexts (kubernetes-contexts--context-names (kubernetes-state)))
+          (context-to-rename (completing-read "Rename context: " contexts)))
+     (list context-to-rename
+           (read-string (format "Rename `%s' to: " context-to-rename)))))
+
+  (let ((contexts (kubernetes-contexts--context-names (kubernetes-state))))
+    (when (not (-contains-p contexts context))
+      (error "Context `%s' does not exist" context-to-rename))
+    (when (-contains-p contexts new-name)
+      (error "Already exists a context named `%s'" new-name)))
+  
+  (-let* (((&alist 'name current-context)
+           (kubernetes-state-current-context (kubernetes-state))))
+    (kubernetes-kubectl
+     kubernetes-props
+     (kubernetes-state)
+     `("config" "rename-context" ,context ,new-name)
+     (lambda (_)
+       (message "Renamed context `%s' to `%s'." context new-name)
+       (kubernetes-contexts-use-context new-name)))))
```

Comment:
`-use-context` should only get called if the context getting renamed is the current context.

Reply:
Done in 0d022ce.